pyensign/ensign.py

```diff
@@ -0,0 +1,173 @@
+import os
+
+from pyensign.connection import Client
+from pyensign.connection import Connection
+from pyensign.auth.client import AuthClient
+from pyensign.exceptions import EnsignResponseType, EnsignTopicCreateError
+
+
+class Ensign:
+    """
+    Ensign connects to an Ensign server and provides access to the Ensign API.
+    """
+
+    def __init__(
+        self,
+        client_id="",
+        client_secret="",
+        endpoint="ensign.rotational.app:443",
+        insecure=False,
+        auth_url="https://auth.rotational.app",
+    ):
+        """
+        Create a new Ensign client with API credentials.
+
+        Parameters
+        ----------
+        client_id : str
+            The client ID part of the API key. If not provided, the client ID is loaded
+            from the ENSIGN_CLIENT_ID environment variable.
+        client_secret : str
+            The client secret part of the API key. If not provided, the client secret
+            is loaded from the ENSIGN_CLIENT_SECRET environment variable.
+        endpoint : str (optional)
+            The endpoint of the Ensign server.
+        insecure : bool (optional)
+            Set to True to use an insecure connection.
+        auth_url : str (optional)
+            The URL of the Ensign authentication server.
+        """
+
+        if not client_id or client_id == "":
+            client_id = os.environ.get("ENSIGN_CLIENT_ID")
+        if not client_secret or client_secret == "":
+            client_secret = os.environ.get("ENSIGN_CLIENT_SECRET")
+        if client_id is None:
+            raise ValueError(
+                "client_id is required but not provided or set in the ENSIGN_CLIENT_ID environment variable"
+            )
+        if client_secret is None:
+            raise ValueError(
+                "client_secret is required but not provided or set in the ENSIGN_CLIENT_SECRET environment variable"
+            )
+        if not isinstance(client_id, str):
+            raise TypeError("client_id must be a string")
+        if not isinstance(client_secret, str):
+            raise TypeError("client_secret must be a string")
+
+        creds = {"client_id": client_id, "client_secret": client_secret}
+        auth = AuthClient(auth_url, creds)
+        connection = Connection(addrport=endpoint, insecure=insecure, auth=auth)
+        self.client = Client(connection)
+
+    def publish(self, events):
```

Comment:
I've been going back and forth on this; on the one hand, you could have the user specify the topic Id in each event as this event signature requires. In the Go SDK we went the other way with a signature that looked like:

```
def publish(self, topic, *events)
```

Then we did a `topic exists` check in the client facing functionality, created the topic if it didn't exist, and fetched the topic ID using a cacheing mechanism, adding each topicId to the event before sending it via gRPC. To prevent confusion about the topicID on the event; we're planning on using an intermediate struct so that all users have to worry about when it comes to events is the raw data and the mimetype. 

Perhaps we want to go that route in pyensign as well?